daemon/image_service.go

```diff
@@ -44,6 +45,11 @@ type ImageService interface {
 	CommitImage(c backend.CommitConfig) (image.ID, error)
 	SquashImage(id, parent string) (string, error)
 
+	// Containerd related methods
+
+	ResolveDescriptor(ctx context.Context, refOrID string) (ocispec.Descriptor, error)
```

Comment:
In my fork we had an [interface](https://github.com/rumpl/moby/blob/c8d/daemon/create.go#L39-L41) that we would [use only in the case we use the containerd store](https://github.com/rumpl/moby/blob/c8d/daemon/create.go#L186) but that sucked.

I realise this isn't ideal, having an interface that does too much and having one of the implementors `panic`... 

I considered moving some of the things over to the `libcontainerd` but I'm not sure it would help and also I'm not really a fan of that wrapper.

Thoughts?

Reply:
The idea is also that this interface will be more and more c8d specific as we move along with the migration.